voip_utils/sip.py

```diff
@@ -72,10 +81,25 @@ def datagram_received(self, data: bytes, addr):
             if not ruri:
                 raise ValueError("Empty receiver URI")
 
+            caller_uri = None
+            caller_name = None
+            # The Contact header should give us the URI used for sending SIP messages to the device
+            if (headers.get('contact') is not None):
+                caller_uri, caller_name = self._parse_uri_header(headers.get('contact'))
+            # We can try using the From header as a fallback
+            elif (headers.get('from') is not None):
+                caller_uri, caller_name = self._parse_uri_header(headers.get('from'))
```

Comment:
Great opportunity to leverage the walrus
```suggestion
            if contact_header := headers.get('contact'):
                caller_uri, caller_name = self._parse_uri_header(contact_header)
            # We can try using the From header as a fallback
            elif from_header := headers.get('from'):
                caller_uri, caller_name = self._parse_uri_header(from_header)
```